tests/lib/bootstrap.js

```diff
@@ -0,0 +1,278 @@
+		"fixedToolbar",
+		"fixedToolbar.backcompat",
+		"fixedToolbar.workarounds",
+		"listview",
+		"listview.autodividers",
+		"listview.backcompat",
+		"listview.hidedividers",
+		"loader",
+		"navbar",
+		"navbar.morebutton",
+		"page",
+		"page.dialog",
+		"page.dialog.backcompat",
+		"pagecontainer",
+		"panel",
+		"popup",
+		"popup.arrow",
+		"popup.arrow.backcompat",
+		"table",
+		"tabs.ajax",
+		"table.reflow",
+		"table.columntoggle",
+		"toolbar",
+		"widget.backcompat",
+		"widget.theme",
+		// Form Widgets
+		"forms/autogrow",
+		"forms/button",
+		"forms/button.backcompat",
+		"forms/checkboxradio",
+		"forms/checkboxradio.backcompat",
+		"forms/clearButton",
+		"forms/flipswitch",
+		"forms/rangeslider",
+		"forms/rangeslider.backcompat",
+		"forms/reset",
+		"forms/select",
+		"forms/select.custom",
+		"forms/slider",
+		"forms/slider.backcompat",
+		"forms/slider.tooltip",
+		"forms/textinput",
+		"forms/textinput.backcompat"
+	];
+
+	var events = [
+		"navigate",
+		"orientationchange",
+		"scroll",
+		"throttledresize",
+		"touch"
+	];
+
+	function getPath( dep ) {
+		for ( var i = 0; i < widgets.length; i++ ) {
+			if ( widgets[i] === dep ) {
+				return "widgets/" + dep;
+			}
+		}
+
+		for ( var i = 0; i < events.length; i++ ) {
+			if ( events[i] === dep ) {
+				return "events/" + dep;
+			}
+		}
+		return dep;
+	}
+
+	function fixPaths( deps ) {
+		for ( var i = 0; i < deps.length; i++ ) {
+			deps[ i ] = getPath( deps[ i ] );
+		}
+
+		return deps;
+	}
+
+	function requireModules( dependencies, noAutoStart, modules ) {
+		if ( !dependencies.length ) {
+
+			$( document ).ready( function() {
+				var $fixture = $( '#qunit-fixture' );
+				if ( $fixture.length ) {
+					QUnit.config.fixture = $fixture.html();
+				}
+
+				if ( !noAutoStart ) {
+					QUnit.start();
+				}
+			} );
+
+		}
+
+		if ( !modules ) {
+			modules = [];
+		}
+
+		var dependency = dependencies.shift();
+
+		if ( dependency ) {
+			require( [ dependency ], function( module ) {
+				modules.push( module );
+				requireModules( dependencies, noAutoStart, modules );
+			} );
+		}
+	}
+
+	( function() {
+
+		var deps = script.getAttribute( "data-deps" );
+
+		if ( deps ) {
+			deps = deps.replace( /^\s+|\s+$/g, "" ).split( /\s+/ );
+		} else {
+			deps = [];
+		}
+
+		deps = fixPaths( deps );
+
+		var full = !!script.getAttribute( "data-full" );
+		var init = !!script.getAttribute( "data-init" );
+		var noBackCompat = !!script.getAttribute( "data-no-backcompat" );
+		var noAutoStart = !!script.getAttribute( "data-no-autostart" );
+		var initAfterModules = !!script.getAttribute( "data-init-after-modules" );
+		var setPushState = !!script.getAttribute( "data-set-push-state" );
+		var setNs = !!script.getAttribute( "data-set-ns" );
+		var overrideEnhanceWithin = !!script.getAttribute( "data-override-enhancewithin" );
+		var modules = script.getAttribute( "data-modules" );
+
+		if ( setPushState ) {
+			deps.concat( [ "jquery-set-push-state" ] );
```

Comment:
why did you prefix these all with jQuery? seems unnecessary since they are only ever used within this file and the names are verbose enough and we have so few dependencies that collisions seem very unlikely